pkg/volume/configmap/configmap.go

```diff
@@ -190,8 +190,9 @@ func (b *configMapVolumeMounter) SetUpAt(dir string, fsGroup *int64) error {
 	return nil
 }
 
-func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap) (map[string][]byte, error) {
-	payload := make(map[string][]byte, len(configMap.Data))
+func makePayload(mappings []api.KeyToPath, configMap *api.ConfigMap) (payload map[string][]byte, mode map[string]int32, err error) {
```

Comment:
Oh, it seems fine to me. But if this `int32` is always there, it will be 0 when not set. Maybe make it `*int32` so the atomic_writer can check if it's set or not, and use the DefaultMode or not?